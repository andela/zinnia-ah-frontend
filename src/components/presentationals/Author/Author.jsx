import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

//utils
import { DEFAULT_USER_PROFILE_IMAGE } from '../../../config/config';

//components
import Avatar from '../Avatar/Avatar';
import Button from '../Button/Button';
import Title from '../Title/Title';

//stylesheets
import '../ProfileSidebar/ProfileSidebar.scss';

const Author = ({ profile }) => {
  return profile ? (
    <div className="container" data-test="author">
      <div className="avatar-div">
        <Link to={`@${profile.username}`}>
          <Avatar
            url={profile.image || DEFAULT_USER_PROFILE_IMAGE}
            className="avatar-lg"
          />
        </Link>
      </div>
      <Title
        content={`${profile.firstName} ${profile.lastName}`}
        className="title-md text-center"
      />
      <Link to={`@${profile.username}`}>
        <p className="text-center username">@{profile.username}</p>
      </Link>
      <div className="d-flex justify-content-center">
        <Button className="btn-white btn-sm" value="FOLLOW" type="submit" />
      </div>
    </div>
  ) : null;
};

Author.propTypes = {
  profile: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
  }),
};

export default Author;
