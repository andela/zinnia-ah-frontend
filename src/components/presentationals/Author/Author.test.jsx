import React from 'react';
import { shallow } from 'enzyme';
import checkPropTypes from 'check-prop-types';

import Author from './Author';

const props = {
  profile: {
    lastName: '',
    firstName: '',
    username: '',
  },
};

const propsWithoutProfile = {};

describe('Author', () => {
  describe('Render', () => {
    it('should render without errors', () => {
      const component = shallow(<Author {...props} />);
      expect(component).toMatchSnapshot();
    });

    it('should render without errors ans without props', () => {
      const component = shallow(<Author {...propsWithoutProfile} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Proptypes', () => {
    it('should receive props', () => {
      const expectedProps = {
        ...props,
      };

      const propsErrors = checkPropTypes(
        Author.propTypes,
        expectedProps,
        'props',
        Author,
      );

      expect(propsErrors).toBeUndefined();
    });
  });
});
