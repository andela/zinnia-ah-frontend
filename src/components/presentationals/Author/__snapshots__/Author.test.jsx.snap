// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Render should render without errors 1`] = `
<div
  className="container"
  data-test="author"
>
  <div
    className="avatar-div"
  >
    <Link
      to="@"
    >
      <Avatar
        className="avatar-lg"
        url="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
      />
    </Link>
  </div>
  <Title
    className="title-md text-center"
    content=" "
  />
  <Link
    to="@"
  >
    <p
      className="text-center username"
    >
      @
    </p>
  </Link>
  <div
    className="d-flex justify-content-center"
  >
    <Button
      className="btn-white btn-sm"
      type="submit"
      value="FOLLOW"
    />
  </div>
</div>
`;

exports[`Author Render should render without errors ans without props 1`] = `""`;
